File: back/node/app/core/database/redis/RedisData.ts
import {RedisClientType} from "redis";
import {GameDao} from "./dao/GameDao.js";

class RedisData {

    private readonly redis: RedisClientType;
    public gameDao: GameDao

    constructor(redis: RedisClientType) {
        this.redis = redis
        this.gameDao = new GameDao(this.redis)
    }

}

export {RedisData}

File: back/kotlin/app/core/database/redis/RedisData.kt
package core.database.redis

import io.lettuce.core.RedisClient
//import io.lettuce.core.api.sync.RedisCommands
import io.lettuce.core.api.async.RedisAsyncCommands
import core.database.redis.dao.GameDao

class RedisData(commands: RedisAsyncCommands<String, String>) : RedisDataProtocol {
    override val gameDao: GameDao = GameDao(commands)
}
